plugins/PostProcessingPlugin/scripts/ChituMods.py

```diff
@@ -0,0 +1,189 @@
+            "key": "ChituMods",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+                "insert_preview_image":
+                {
+                    "label": "Insert a preview image for printers with chitu boards?",
+                    "description": "Selecting this a image will be generated and included into the gcode so the touch displays can show it",
+                    "type": "bool",
+                    "default_value": false
+                },
+                "insert_time_info":
+                
+                {
+                    "label": "Insert the needed time and the elapsed time?",
+                    "description": "Selecting this the calculated time and the elapsed time will be included in chitu style",
+                    "type": "bool",
+                    "default_value": false
+                }
+            }
+        }"""
+
+    def execute(self, in_data):
+        # we get a list, each list item is the command set for a complete layer
+        self.out_data = in_data
+        if self.getSettingValueByKey("insert_preview_image"):
+            self.img_data=[]
+            self._createSnapshot()
+            self.img_data.append('\n'.join(self.generate_image_code(self._snapshot))) # create one long string and add it as item to a list
+            self.img_data[0] += ('\n') # add an additional newline, looks better
+            self.img_data.extend(in_data)
+            self.out_data=self.img_data      
+        else:
+            self.out_data=in_data
+
+        if self.getSettingValueByKey("insert_time_info"):
+            Logger.log("d", "Modifying time info for chitu ...")
+            time_data=self.insert_time_infos(self.out_data)
+            self.out_data=time_data
+        return self.out_data
+    
+    
+
+    def insert_time_infos(self, gcode_data):
+        return_data=[]
+        Logger.log("d", "Modifying time info for chitu ...")
+        for gcode in gcode_data:
+            lines = gcode.split('\n')
+            modified = False
+            for index, line in enumerate(lines):
+                if line.startswith(';TIME:'):
+                    lines[index] = 'M2100 T%d' % int(getValue(line, ';TIME:', 0))
+                    modified=True
+                elif line.startswith(';TIME_ELAPSED:'):
+                    lines[index] = 'M2101 T%d' % int(getValue(line, ';TIME_ELAPSED:', 0)) 
+                    modified=True
+            if modified:
+                return_data.append('\n'.join(lines))
+            else:
+                return_data.append(gcode)
+
+        return return_data        
+        
+
+    def _createSnapshot(self, *args):
+        Logger.log("d", "Creating tronxy thumbnail image ...")
+        try:
+            self._snapshot = Snapshot.snapshot(width = 300, height = 300)
+        except Exception:
+            Logger.logException("w", "Failed to create snapshot image")
+            self._snapshot = None  

```

Comment:
instead of storing the _snapshot in the class instance, it would be nicer to return it from this method.

Reply:
Done.